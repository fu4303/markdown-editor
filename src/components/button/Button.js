import React from "react"
import PropTypes from "prop-types"
import styled from "@emotion/styled"

const Button = ({ text, action }) => (
  <ButtonWrapper onClick={action}>{text}</ButtonWrapper>
)

Button.propTypes = {
  text: PropTypes.string.isRequired,
  action: PropTypes.func,
}

Button.defaultProps = {
  action: null,
}

export default Button

const ButtonWrapper = styled.button`
  background: var(--light);
  border: none;
  border-radius: var(--xxs);
  padding: var(--xxs) var(--sm);
  font-size: var(--xs);
  font-family: "Roboto", sans-serif;
  font-weight: 500;
  cursor: pointer;
  color: var(--primary);
  -webkit-appearance: none;
  transition: transform 0.1s ease-in-out;
  outline: none;
  :hover {
    transform: scale(1.05);
  }
  :active {
    transform: scale(0.95);
  }
`
