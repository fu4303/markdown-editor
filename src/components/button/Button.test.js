import React from "react"
import { shallow } from "enzyme"
import toJson from "enzyme-to-json"
import Button from "./Button"

describe("<Button />", () => {
  it("renders without crashing", () => {
    shallow(<Button text="button prop text" />)
  })

  it("matches snapshot with data passed correctly", () => {
    const wrapper = shallow(
      <Button text="button prop text" action={jest.fn()} />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it("on button click calls action function", () => {
    const actionMock = jest.fn()
    const wrapper = shallow(
      <Button text="button prop text" action={actionMock} />
    )
    wrapper.simulate("click")
    expect(actionMock).toHaveBeenCalled()
  })
})
