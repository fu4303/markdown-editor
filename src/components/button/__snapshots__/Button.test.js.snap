// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> matches snapshot with data passed correctly 1`] = `
<Styled(button)
  onClick={[MockFunction]}
>
  button prop text
</Styled(button)>
`;
